src/models/dict_model.jl

```diff
@@ -2,6 +2,7 @@ mutable struct DictModel <: AbstractModel
     objective::Union{Nothing, Objective}
     eq_constraints::VectorOfFunctions
     ineq_constraints::VectorOfFunctions
+    sd_function::Union{AbstractFunction, Nothing}
```

Comment:
Let's have this called `sd_constraints`. We should allow multiple such constraints.